k8s/k8s.go

```diff
@@ -0,0 +1,182 @@
+			return nil, err
+		}
+
+		// Create the endpoint object and configure it
+		endpoint = &libcalico.Endpoint{ProfileIDs: []string{profileID}, OrchestratorID: orchestratorID,
+			WorkloadID: workloadID, Hostname: hostname, State: "active"}
+		if err = utils.PopulateEndpointNets(endpoint, result); err != nil {
+			return nil, err
+		}
+
+		labels, err := getK8sLabels(client, k8sArgs)
+		if err != nil {
+			return nil, err
+		}
+		endpoint.Labels = labels
+
+		fmt.Fprintf(os.Stderr, "Calico CNI using IPv4=%s IPv6=%s\n", endpoint.IPv4Nets, endpoint.IPv6Nets)
+	}
+
+	// Whether the endpoint existed or not, the veth needs recreating.
+	hostVethName, contVethMac, err := utils.DoNetworking(args, conf, result)
+	if err != nil {
+		return nil, err
+	}
+	endpoint.Mac = contVethMac
+	endpoint.Name = hostVethName
+
+	// Write the endpoint object (either the newly created one, or the updated one)
+	if err := endpoint.Write(etcd); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func newK8sClient(conf utils.NetConf) (*k8sclient.Client, error) {
+	// Some config can be passed in a kubeconfig file
+	kubeconfig := conf.Kubernetes.Kubeconfig
+
+	// but that config can be overridden by config passed in explicitly in the network config.
+	configOverrides := &clientcmd.ConfigOverrides{}
+	var overridesMap = []struct {
+		variable *string
+		value    string
+	}{
+		{&configOverrides.ClusterInfo.Server, conf.Policy.K8sAPIRoot},
+		{&configOverrides.AuthInfo.ClientCertificate, conf.Policy.K8sClientCertificate},
+		{&configOverrides.AuthInfo.ClientKey, conf.Policy.K8sClientKey},
+		{&configOverrides.ClusterInfo.CertificateAuthority, conf.Policy.K8sCertificateAuthority},
+		{&configOverrides.AuthInfo.Username, conf.Policy.K8sUsername},
+		{&configOverrides.AuthInfo.Password, conf.Policy.K8sPassword},
+	}
+
+	// Using the override map above, populate any non-empty values.
+	for _, override := range overridesMap {
+		if override.value != "" {
+			*override.variable = override.value
+		}
+	}
+
+	// Use the kubernetes client code to load the kubeconfig file and combine it with the overrides.
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfig},
+		configOverrides).ClientConfig()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return k8sclient.New(config)
+}
+
+func getK8sLabels(client *k8sclient.Client, k8sargs utils.K8sArgs) (map[string]string, error) {
+	pods, err := client.Pods(string(k8sargs.K8S_POD_NAMESPACE)).Get(fmt.Sprintf("%s", k8sargs.K8S_POD_NAME))
+	if err != nil {
+		return nil, err
+	}
+
+	labels := pods.Labels
+	labels["calico/k8s_ns"] = fmt.Sprintf("%s", k8sargs.K8S_POD_NAMESPACE)
+	return labels, nil
+}
+
+func getPodCidr(client *k8sclient.Client, k8sargs utils.K8sArgs, conf utils.NetConf, hostname string) (podCidr string, err error) {
+	nodeName := hostname
+
+	if conf.Kubernetes.NodeName != "" {
+		nodeName = conf.Kubernetes.NodeName
+	}
+	var node *api.Node
```

Comment:
I don't think you need to pre-declare the node var - it's only used within the `if` block below.